internal/api_server/middleware/tls.go

```diff
@@ -27,8 +27,17 @@ func NewHTTPServer(router http.Handler, log logrus.FieldLogger, address string)
 	}
 }
 
-func NewHTTPServerWithTLSContext(router http.Handler, log logrus.FieldLogger, address string) *http.Server {
-	server := NewHTTPServer(router, log, address)
+func NewAgentHTTPServer(router http.Handler, log logrus.FieldLogger, address string) *http.Server {
+	return &http.Server{
+		Addr:              address,
+		Handler:           router,
```

Comment:
We need different settings on the agent side, please note this happens after TLS has validated client certificate. At that time we have a bigger Idle Timeout... to allow console, and We limit the Read time only to headers.